Sources/TensorFlow/Operators/NN.swift

```diff
@@ -820,3 +820,45 @@ func _vjpAvgPool3D<Scalar: TensorFlowFloatingPoint>(
     }
   )
 }
+
+//===------------------------------------------------------------------------------------------===//
+// Rearrange depth/space
+//===------------------------------------------------------------------------------------------===//
+
+/// Rearranges data from depth into blocks of spatial data.
```

Comment:
I'm flabbergasted to find I named it `size`😂
It must be `blockSize`!

> this would be much more idiomatic Swift as an extension method

I thought about it but this function is only for tensors with rank 4.
I think it's not good to have a method that cannot be used on most of tensors.

Reply:
Hmm.  My first reaction is, “oh, that makes sense.”  But then after thinking more I see no fundamental difference between a method with preconditions and a function with preconditions.  Is your point that it might just be too easy to find an inapplicable method via code completion?

I'm working on a formulation that captures rank in the tensor type, so eventually we can solve these problems with conditional extensions.